### PR TITLE
Add uniform water condition

### PR DESCRIPTION
**Pull request by aronnoordam**

### CODE REVIEW

File: stem/IO/kratos_water_processes_io.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            - domain (str): name of the Kratos domain
+
+        """
+        self.domain = domain
+
+    def __create_uniform_water_pressure_dict(
+        self, part_name: str, parameters: UniformWaterPressure
+    ) -> Dict[str, Any]:
+        """
+        Creates a dictionary containing the absorbing boundary parameters
```
Comment (brunozc) [inline stem/IO/kratos_water_processes_io.py:31]: it is not absorbing BC

File: stem/IO/kratos_water_processes_io.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        Args:
+            - part_name (str): part name where the boundary condition is applied
+            - parameters (:class:`stem.boundary.AbsorbingBoundary`): absorbing boundary parameters object
```
Comment (brunozc) [inline stem/IO/kratos_water_processes_io.py:35]: it is not abs

File: stem/water_processes.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from dataclasses import dataclass
+from abc import ABC
+
+
+@dataclass
+class WaterProcessParametersABC(ABC):
+    """
+    Abstract class which contains the parameters for a water process.
+    """
+
```
Comment (brunozc) [inline stem/water_processes.py:10]: pass

File: stem/water_processes.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class UniformWaterPressure(WaterProcessParametersABC):
+    """
+    Class which contains the parameters for a uniform water pressure process.
+
+    Inheritance:
+        - WaterProcessParametersABC (ABC): Abstract base class which contains the parameters for a water process.
+
+    Attributes:
+        - water_pressure (float): The water pressure.
+        - is_fixed (bool): Whether the water pressure is fixed or not (default: True).
```
Comment (brunozc) [inline stem/water_processes.py:22]: is fixed you mean is constant?

Comment (aronnoordam) [reply stem/water_processes.py:22]: it cannot be changed during calculation, thus no water pressures are calculated during the kratos calculation

**Review by brunozc — APPROVED** [review]

**Pull request merged**